libraries/AP_HAL_ChibiOS/RCOutput_bdshot.cpp

```diff
@@ -458,9 +504,17 @@ __RAMFUNC__ void RCOutput::bdshot_finish_dshot_gcr_transaction(virtual_timer_t*
     }
 #endif
 
-    // rotate to the next input channel
     group->bdshot.prev_telem_chan = group->bdshot.curr_telem_chan;
+    // rotate to the next input channel, we have to rotate even on failure otherwise
+    // we will not get data from active channels
     group->bdshot.curr_telem_chan = bdshot_find_next_ic_channel(*group);
+
+    if (group->bdshot.dma_tx_size > 0) {
+        group->dshot_state = DshotState::RECV_COMPLETE;
+    } else {
+        group->dshot_state = DshotState::RECV_FAILED;
```

Comment:
comment worthwhile on why this is needed - when no EDT reply?

Reply:
Fixed